src/main/java/core/basesyntax/io/impl/CsvTransactionRequestDeserializer.java

```
@@ -0,0 +1,39 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CSVTransactionRequestDeserializer implements TransactionRequestDeserializer {
+    private final FruitTypeFactory fruitTypeFactory;
+
+    public CSVTransactionRequestDeserializer(FruitTypeFactory fruitTypeFactory) {
+        this.fruitTypeFactory = fruitTypeFactory;
+    }
+
+    @Override
+    public List<TransactionRequest> loadFrom(Path path) {
+        try (BufferedReader reader = Files.newBufferedReader(path)) {
+            return reader.lines().skip(1).map(this::parseLine).collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

This violates checklist item: "Don't ignore exceptions in try-catch blocks\nDon't leave `e.printStackTrace()` or `System.out.println(e.getMessage())` in the catch blocks.\nLet's rethrow a RuntimeException with an **informative** message and exception object." Please include an informative message and the original cause when rethrowing. For example: `throw new RuntimeException("Failed to read file: " + path, e);` instead of `throw new RuntimeException(e);` (current catch at lines 25-26).